source/fundamentals/connection.txt

```diff
@@ -0,0 +1,434 @@
+DNS SRV record, use ``mongodb`` to specify the :manual:`Standard Connection
+Format </reference/connection-string/#standard-connection-string-format>`.
+
+.. note::
+
+   If your deployment is on MongoDB Atlas, follow the
+   :atlas:`Atlas driver connection guide <driver-connection?jmp=docs_driver_java>`
+   to retrieve your connection string.
+
+The next part of the connection string contains your username and password
+if you are using password-based authentication. Replace the value of ``user``
+with your username and ``pass`` with your password. If you are using an
+authentication mechanism that does not require a username and password, omit
+this part of the connection URI.
+
+The next part of the connection string specifies the hostname or IP address and
+port of your MongoDB instance. In the example below, we use ``sample-hostname``
+as the hostname and ``27017`` as the port. Replace these values to point to
+your MongoDB instance.
+
+The last part of the connection string contains connection and authentication
+options as parameters. In the example below, we set two connection options:
+``retryWrites=true`` and ``w=majority``. For more information on connection
+options, skip to the :ref:`connection options <connection-options>` section.
+
+The code below shows how you can use the sample connection URI in a client to
+connect to MongoDB.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/srv.java
+   :language: java
+
+.. _connect-replica-set:
+
+Connect to a Replica Set
+------------------------
+
+A MongoDB replica set deployment is a group of connected instances that
+store the same set of data. This configuration of instances provides data
+redundancy and high data availability.
+
+To connect to a replica set deployment, specify the hostname (or IP address)
+and port numbers of one or more members of the replica set separated by commas.
+Use the :manual:`replica set option
+</reference/connection-string/#urioption.replicaSet>` to specify the name
+of the replica set with which your application is connecting. When the
+replica set option is specified, the driver assumes that your
+connection string uses a seed list of multiple hosts instead of a single
+``host:port`` configuration.
+
+The connection string below specifies three hosts in the cluster and
+replica set named "myRs".
+
+.. code-block:: none
+
+   mongodb://host1:27017,host2:27017,host3:27017/?replicaSet=myRs
+
+.. note::
+
+   Although it is possible to connect to a replica set deployment by
+   providing only a single host, you should provide the driver with the
+   full list to ensure that it is able to connect even if one host fails.
+
+.. _compression:
+
+Compression
+-----------
+
+You can compress messages passing between your MongoDB instance and the
+driver. MongoDB drivers support up to three different algorithms
+depending on release version:
+
+1. `Snappy <https://google.github.io/snappy/>`__: available in MongoDB 3.4 and later.
```

Comment:
Perhaps make a note here that snappy & zstandard require an extra dependency?

Reply:
There is a subsection that explains this below -- do you think we need more than that?

Reply:
I'm probably being over zealous that someone will miss that.